### DISCUSSION TITLE
Generation of structured multiblock mesh

### DISCUSSION THREAD

**Discussion by EdoAlvarezR** [mdolab/dafoam#541] [General]
Hi all,

I am looking at setting up a 2D axisymmetric problem of a ducted fan with a duct, centerbody, and an actuator disk (fan stage). My guess is that using multiple structured blockmeshes would help reduce numerical noise:

![New Drawing 16](https://github.com/mdolab/dafoam/assets/14857108/41792a1c-2538-446f-a4a1-f9b171caee33)

Is this the right approach to meshing this problem? What tools would you recommend for generating the multiblock mesh?

So far I've been thinking of generating blocks 1 and 2 with pyHyp, and then coding my own tool for generating and connecting blocks 3, 4, 5, and 6 and pass the whole thing to DAFoam through `blockMeshDict`. Any feedback, ideas, or recommendations would be appreciated!

**Comment by friedenhe** [comment]
The mesh topology looks good to me. Although I am not sure how blockMeshDict can connect these blocks... We have been using Pointwise to generate unstructured meshes, it can generate very refined boundary layer meshes and maintain good mesh quality. We are not aware of a good open-source mesh generation tool that works well for our cases.. SnappyHexMesh has some problems refining the boundary layer.
